pkg/controller/servicebinding/binder.go

```diff
@@ -114,13 +113,27 @@ func (b *binder) getApplicationByLabelSelector() (*unstructured.UnstructuredList
 
 // extractSpecVolumes based on volume path, extract it unstructured. It can return error on trying
 // to find data in informed Unstructured object.
-func (b *binder) extractSpecVolumes(obj *unstructured.Unstructured) ([]interface{}, error) {
+func (b *binder) extractSpecVolumes(obj *unstructured.Unstructured) ([]corev1.Volume, error) {
+	var volumes []corev1.Volume
+
 	log := b.logger.WithValues("Volumes.NestedPath", b.getVolumesPath())
 	log.Debug("Reading volumes definitions...")
-	volumes, _, err := unstructured.NestedSlice(obj.Object, b.getVolumesPath()...)
+
+	volumesUnstructured, _, err := unstructured.NestedSlice(obj.Object, b.getVolumesPath()...)
 	if err != nil {
 		return nil, err
 	}
+
+	for _, v := range volumesUnstructured {
+		var volume corev1.Volume
```

Comment:
I believe you could `json.Unmarshal` the slice instead of each element of the slice (untested):

```go
var volumes []corev1.Volume
err = json.Unmarshal(volumesUnstructured, &volumes)
if err != nil { ... }
```